analyzer/codechecker_analyzer/analyzers/clangtidy/analyzer.py

```diff
@@ -267,19 +308,44 @@ def construct_config_handler(cls, args, context):
             # No clang tidy arguments file was given in the command line.
             LOG.debug_analyzer(aerr)
 
-        try:
-            # The config file dumped by clang-tidy contains "..." at the end.
-            # This has to be emitted, otherwise -config flag of clang-tidy
-            # cannot consume it.
-            with open(args.tidy_config, 'rb') as tidy_config:
-                lines = tidy_config.readlines()
-                lines = [x for x in lines if x != '...\n']
-                handler.checker_config = ''.join(lines)
-        except IOError as ioerr:
-            LOG.debug_analyzer(ioerr)
-        except AttributeError as aerr:
-            # No clang tidy config file was given in the command line.
-            LOG.debug_analyzer(aerr)
+        analyzer_config = {}
+        # TODO: This extra "isinsrance" check is needed for
+        # CodeChecker analyzers --analyzer-config. This command also
+        # runs this function in order to construct a config handler.
+        if 'analyzer_config' in args and \
+                isinstance(args.analyzer_config, list):
+            r = re.compile(r'(?P<analyzer>.+?):(?P<key>.+?)=(?P<value>.+)')
+            for cfg in args.analyzer_config:
+                m = re.search(r, cfg)
+                if m.group('analyzer') == cls.ANALYZER_NAME:
+                    analyzer_config[m.group('key')] = m.group('value')
+
+        # TODO: This extra "isinsrance" check is needed for
+        # CodeChecker checkers --checker-config. This command also
+        # runs this function in order to construct a config handler.
+        if 'checker_config' in args and \
+                isinstance(args.checker_config, list):
+            r = re.compile(r'(?P<analyzer>.+?):(?P<key>.+?)=(?P<value>.+)')
+            check_options = []
+
+            for cfg in args.checker_config:
+                m = re.search(r, cfg)
+                if m.group('analyzer') == cls.ANALYZER_NAME:
+                    check_options.append({'key': m.group('key'),
+                                          'value': m.group('value')})
+            analyzer_config['CheckOptions'] = check_options
+        else:
+            try:
+                with open(args.tidy_config, 'r') as tidy_config:
```

Comment:
The following arguments are missing from the open call: `encoding="utf-8", errors="ignore"`